src/extractor/edge_based_graph_factory.cpp

```diff
@@ -426,38 +428,27 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                     continue;
                 }
 
-                if (is_barrier_node)
-                {
-                    if (node_u != node_w)
-                    {
-                        ++skipped_barrier_turns_counter;
-                        continue;
-                    }
-                }
-                else
```

Comment:
Not omitting turns through barriers breaks parts of the routability checks in cucumber.